java/core/src/java/org/apache/orc/impl/StringHashTableDictionary.java

```diff
@@ -173,10 +170,18 @@ public int size() {
 
   /**
    * Compute the hash value and find the corresponding index.
-   *
    */
   int getIndex(Text text) {
-    return Math.floorMod(text.hashCode(), capacity);
+    return getIndex(text.getBytes(), 0, text.getLength());
+  }
+
+  /**
+   * Compute the hash value and find the corresponding index. Uses same
+   * implementation as {@code Text#hashCode()}.
+   */
+  int getIndex(final byte[] bytes, final int offset, final int length) {
+    return Math.floorMod(WritableComparator.hashBytes(bytes, offset, length),
```

Comment:
Now that there is an implicit dependency on the hash function of `Test#hashCode` and this class, is there any way to enforce that a change in the hash function would reflect here as well?

Reply:
Thats exactly what I had in mind with my comment above..

Reply:
I am happy to sub this out with non-hadoop hashcode if you would like.  Like I said, I came up with this only to preserve the previous behavior.